lib/services/network/backend_service.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:bluebubbles/models/models.dart';
+import 'package:bluebubbles/services/rustpush/rustpush_service.dart';
+import 'package:bluebubbles/services/services.dart';
+import 'package:dio/dio.dart';
+import 'package:get/get.dart';
+
+BackendService backend = Get.isRegistered<BackendService>() ? Get.find<BackendService>() : Get.put(RustPushBackend());
+enum ParticipantOp { Add, Remove }
+
+abstract class BackendService {
```

Comment:
Don't you need to extend GetxService here? And then your init() at the bottom gets an override annotation so it runs when the service is init

Reply:
The way I did it was I implemented this as an interface. Maybe it would be a good idea to extend getx then have services extend this? I dunno I kinda hate the whole OOP java thing